### PR TITLE
Changed unit entity tracker to use map instead of array to avoid collisions

### PR DESCRIPTION
**Pull request by Marsunpaisti**
Unit entities stored in a hashmap instead of in an array by unitindexes, fixed error msg on lesser hydra death. Also .w3x changes for hydra items.

### CODE REVIEW

File: wurst/systems/units/UnitEntity.wurst [hunk 1]
```diff
@@ -50,7 +53,9 @@ public abstract class UnitEntity
         postCreate()
 
     function postCreate()
-        instances[theUnit.getIndex()] = this
+        if instances.has(theUnit)
```
Comment (jonathanmclaus) [inline wurst/systems/units/UnitEntity.wurst]: This is weirdly formatted.

File: wurst/systems/units/UnitEntity.wurst [hunk 2]
```diff
@@ -11,7 +11,7 @@ public function randomAngle() returns angle
 
 
 public abstract class UnitEntity
-    static thistype array instances
+    static HashMap<unit, thistype> instances = new HashMap<unit,thistype>()
```
Comment (jonathanmclaus) [inline wurst/systems/units/UnitEntity.wurst]: What is the overhead on this?

File: wurst/systems/units/UnitEntity.wurst (hunk 1 above)
Comment (Marsunpaisti) [reply wurst/systems/units/UnitEntity.wurst]: Leftovers from debug code

File: wurst/systems/units/UnitEntity.wurst (hunk 2 above)
Comment (Marsunpaisti) [reply wurst/systems/units/UnitEntity.wurst]: No idea to be honest but I'd like to try it to see if its more reliable than using unitindex

File: wurst/systems/units/hostiles/LesserHydra.wurst [hunk 3]
```diff
@@ -95,9 +94,8 @@ public class LesserHydra extends Hostile
 
     override function onDeath()
         deathPos = getPos()
-        removeUnit()
```
Comment (theQuazz) [inline wurst/systems/units/hostiles/LesserHydra.wurst]: We definitely want to remove the unit here

Comment (Marsunpaisti) [reply wurst/systems/units/hostiles/LesserHydra.wurst]: Are you sure about that? The ondestroy function removes the unit later

**Pull request merged**